### PR TITLE
SW-4440 Validate photo files at upload time

### PR DESCRIPTION
**Pull request by sgrimm**
To prevent clients from uploading photos in unsupported image formats, try loading
the image data as if we were about to generate a thumbnail for it. If that fails,
delete the uploaded file and return an error.

### CODE REVIEW

**Comment by sgrimm** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1464** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1464?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #1465** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1465?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #1466** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1466?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1466?utm_source=stack-comment).

File: src/main/kotlin/com/terraformation/backend/report/ReportFileService.kt [hunk 1]
```diff
@@ -168,7 +168,7 @@ class ReportFileService(
   ): FileId {
     requirePermissions { updateReport(reportId) }
 
-    val fileId = fileService.storeFile("report", data, metadata, insertChildRow)
+    val fileId = fileService.storeFile("report", data, metadata, null, insertChildRow)
```
Comment (nickgraz) [inline src/main/kotlin/com/terraformation/backend/report/ReportFileService.kt:171]: Is there a reason these files don't need to be validated?

**Review by nickgraz — APPROVED** [review]
One question, but otherwise looks good

File: src/main/kotlin/com/terraformation/backend/report/ReportFileService.kt (hunk 1 above)
Comment (sgrimm) [reply src/main/kotlin/com/terraformation/backend/report/ReportFileService.kt:171]: Files attached to reports aren't necessarily images; they could be Excel spreadsheets or Word documents.

**Comment by sgrimm** [comment]
Holding off merging this for now pending some more investigation on the mobile app side (see SW-4441 for details).

**Pull request merged**